wallet/how-to/use-non-evm-networks/starknet/create-a-simple-starknet-dapp.md

```diff
@@ -132,27 +170,82 @@ This provides visual feedback to the user, confirming which wallet they are usin
 The following code is an example of how to update the interface with the connected wallet's details:
 
 ```typescript title="App.tsx"
-import { useState } from "react";
-
 function App() {
-  const [walletName, setWalletName] = useState("");
-  const [walletAddress, setWalletAddress] = useState("");
-  const [walletIcon, setWalletIcon] = useState("");
+  const [walletName, setWalletName] = useState("")
+  const [walletAddress, setWalletAddress] = useState("")
+  const [walletIcon, setWalletIcon] = useState("")
+  const [walletAccount, setWalletAccount] = useState<AccountInterface | null>(null)
 
+  // Connect the handler.
   async function handleConnect(options?: ConnectOptions) {
-    const res = await connect(options);
-    setWalletName(res?.name || "");
-    setWalletAddress(res?.account?.address || "");
-    setWalletIcon(res?.icon || "");
+    const res = await connect(options)
+    setWalletName(res?.name || "")
+    setWalletAddress(res?.account?.address || "")
+    setWalletIcon(res?.icon || "")
+    setWalletAccount(res?.account)
+  }
+
+  // Disconnect the handler.
+  async function handleDisconnect(options?: DisconnectOptions) {
+    await disconnect(options)
+    setWalletName("")
+    setWalletAddress("")
+    setWalletAccount(null)
   }
 
   return (
-    <div>
-      <h2>Selected Wallet: {walletName}</h2>
-      <p>Address: {walletAddress}</p>
-      <img src={walletIcon} alt="Wallet icon" />
+    <div className="App">
+      <h1>get-starknet</h1>
+      <div className="card">
+        <button onClick={() => handleConnect()}>
+          Default
+        </button>
+        <button onClick={() => handleConnect({ modalMode: "alwaysAsk" })}>
+          Always ask
+        </button>
+        <button onClick={() => handleConnect({ modalMode: "neverAsk" })}>
+          Never ask
+        </button>
+        <button
+          onClick={() =>
+            handleConnect({
+              modalMode: "alwaysAsk",
+              modalTheme: "dark",
+            })
+          }
+        >
+          Always ask with dark theme
+        </button>
+        <button
+          onClick={() =>
+            handleConnect({
+              modalMode: "alwaysAsk",
+              modalTheme: "light",
+            })
+          }
+        >
+          Always ask with light theme
+        </button>
+        <button onClick={() => handleDisconnect()}>
+          Disconnect
+        </button>
+        <button onClick={() => handleDisconnect({ clearLastWallet: true })}>
+          Disconnect and reset
+        </button>
+      </div>
+      {walletName && (
+        <div>
+          <h2>
+            Selected Wallet: <pre>{walletName}</pre>
+            <img src={walletIcon} alt="Wallet icon"/>
+          </h2>
+          <ul>
+            <li>Wallet address: <pre>{walletAddress}</pre></li>
+          </ul>
+        </div>
+      )}
     </div>
-  );
+  )
 }
 ```
 
```

Comment:
Suggestion: For the full example, maybe just choose one connect option and one disconnect option to display, to minimize confusion and make it feel more like a realistic example?

Reply:
How would you like to present this? I think part of this was to highlight the modal mode and modal theme, to show what can be done.